### DISCUSSION TITLE
Error at generateFunctionFromString since version 3.13.0

### DISCUSSION THREAD

**Discussion by jonnitto** [alpinejs/alpine#3726] [7. Bugs (archived)]
Since AlpineJS version `3.13.0` I'got following error in the console:

```
Cannot read properties of null (reading 'trim')
```

It looks like it happens in this function:
https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/evaluator.js#L71-L77

**Comment by ekwoka** [comment]
Did you figure out the issue? or no?

Is there a specific directive you used that caused this?

This should only be an issue, if you used a directive that requires an expression but didn't provide the expression. So it would be helpful to know the directives you used that caused this to make sure that either they don't actually require an expression and should do a null check, or that they do require one and it may need to be made more clear.

**Reply by jonnitto** [reply]
The issue was in indeed in my code base who was not working anymore. That's why I closed this, as it was a fault in my code. It was working before, but it was not 100% clean.

**Comment by loilo** [comment]
Additional hint for possible future searchers (like me): In my case, the issue stemmed from me accidentally passing `undefined` to the [`$watch`](https://alpinejs.dev/magics/watch) function.
